### ISSUE TITLE
Flutter Android APP network issue while switching connection from wi-fi to 4G

### ISSUE THREAD

**Issue by qtldleon** [flutter/flutter#139898]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

Write a very simple Flutter Android, just one page, use "Image.network()" to display a network image. (see code sample section)

1. Quit the app which mentioned above
2. turn on Android phone's wi-fi, make sure it connects to a router which connected to internet. Now you phone should be able to access internet via wi-fi.
3. Start the app which mentioned above
4. Turn off wi-fi, wait few seconds, now your phone should be able to access intetnet via cellular
5. Click the "Show Image" button on the page

### Expected results

After click the "Show Image" button, the image should be displayed under the "Show Image" button

### Actual results

After click the "Show Image" button, there is SocketException occured, and the image is not displayed. See below logs for details:

SocketException: Connection failed (OS Error: Machine is not on the network, errno = 64), address = img1.utoofishing.com, port = 443

### Code sample

<details><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

class DebugNetworkChangePage extends StatefulWidget {
  @override
  _DebugNetworkChangePageState createState() => _DebugNetworkChangePageState();
}

class _DebugNetworkChangePageState extends State<DebugNetworkChangePage> {
  String imageUrl = '';

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text('Test Network')),
      body: Padding(
        padding: EdgeInsets.all(10),
        child: Column(
          children: [
            ElevatedButton(
              child: Text('Show Image'),
              onPressed: () {
                // could change this to any image url
                imageUrl =
                    'https://img1.utoofishing.com/P01/asset/M/X/1701822719nVxzsQkbetNWc1MX.jpeg';
                setState(() {});
              },
            ),
            if (imageUrl.isNotEmpty) Image.network(imageUrl),
          ],
        ),
      ),
    );
  }
}

```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
➜  ~ flutter doctor -v
[✓] Flutter (Channel stable, 3.13.9, on macOS 14.1.2 23B92 darwin-arm64, locale
    zh-Hans-CN)
    • Flutter version 3.13.9 on channel stable at /opt/leon/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d211f42860 (7 weeks ago), 2023-10-25 13:42:25 -0700
    • Engine revision 0545f8705d
    • Dart version 3.1.5
    • DevTools version 2.25.0
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /opt/leon/android
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /opt/leon/android
    • ANDROID_SDK_ROOT = /opt/leon/android
    • Java binary at: /Applications/Android
      Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✗] Xcode - develop for iOS and macOS
    ✗ Xcode installation is incomplete; a full installation is necessary for iOS
      and macOS development.
      Download at: https://developer.apple.com/xcode/download/
      Or install Xcode via the App Store.
      Once installed, run:
        sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer
        sudo xcodebuild -runFirstLaunch
    ✗ CocoaPods not installed.
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin
        code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To install see
      https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b829.9-10027231)

[✓] IntelliJ IDEA Ultimate Edition (version 2023.2.5)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.84.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (3 available)
    • ADT AN00 (mobile) • A63Q6R3511000455 • android-arm64  • Android 12 (API
      31)
    • macOS (desktop)   • macos            • darwin-arm64   • macOS 14.1.2 23B92
      darwin-arm64
    • Chrome (web)      • chrome           • web-javascript • Google Chrome
      119.0.6045.199

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
➜  ~
```

</details>

**Comment by qtldleon** [comment]
This issue only happenes when switch connection from wi-fi to 4G (cellular data). If I use 4G before start APP, then switch to wi-fi after started APP, the network functionality works fine.

**Comment by qtldleon** [comment]
For above reproduce steps, after step 5 (Click the "Show Image" button on the page), there is network issue. But if switch connection back to wi-fi, and click the "Show Image" button again, I can see the image displayed.

**Comment by dam-ease** [comment]
Thanks for filing this.
I'm unable to reproduce this on my end and the image came up when I switched from WiFi to 4G.

![Screenshot_20231211-135200](https://github.com/flutter/flutter/assets/53122008/607d3ba9-0e99-4e1a-8ac1-fbc35ec49672)

Can you test this out on the latest versions of Flutter? I see you are on an older version.

<details><summary>flutter doctor -v</summary>
<p>

```
[!] Flutter (Channel stable, 3.16.3, on macOS 14.0 23A344 darwin-arm64, locale
    en-NG)
    • Flutter version 3.16.3 on channel stable at
      /Users/damilolaalimi/sdks/flutter
    ! Warning: `dart` on your path resolves to
      /opt/homebrew/Cellar/dart/3.1.5/libexec/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/damilolaalimi/sdks/flutter.
      Consider adding /Users/damilolaalimi/sdks/flutter/bin to the front of your
      path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b0366e0a3f (5 days ago), 2023-12-05 19:46:39 -0800
    • Engine revision 54a7145303
    • Dart version 3.2.3
    • DevTools version 2.28.4
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/damilolaalimi/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/damilolaalimi/Library/Android/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b802.4-9586694)

[!] Android Studio (version unknown)
    • Android Studio at /Users/damilolaalimi/Downloads/Android Studio
      Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build
      17.0.7+0-17.0.7b1000.6-10550314)

[✓] VS Code (version 1.84.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.50.0

[✓] VS Code (version 1.83.1)
    • VS Code at /Users/damilolaalimi/Downloads/Visual Studio Code.app/Contents
    • Flutter extension version 3.50.0

[✓] Connected device (4 available)
    • TECNO CC7 (mobile)         • 05199339BL004045          • android-arm64  •
      Android 9 (API 28)
    • Damilola’s iPhone (mobile) • 00008110-001964480AE1801E • ios            •
      iOS 17.1.1 21B91
    • macOS (desktop)            • macos                     • darwin-arm64   •
      macOS 14.0 23A344 darwin-arm64
    • Chrome (web)               • chrome                    • web-javascript •
      Google Chrome 120.0.6099.71

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.
``` 

</p>
</details>

**Comment by qtldleon** [comment]
I will try latest Flutter later. Meantime, please check if you followed the reproduce steps exactly. If not, you may want try it again, but you need another image url, because that one is already cached. 

change the image url to https://img2.utoofishing.com/P02/asset/A/T/img_z85T0kHxEW6ayxnwWXto0Q7nuSTYHOAT.jpeg in your code.

This issue only happened in below case: start APP when it's wi-fi connection, then turn off wi-fi connection, then all network related operation fails.

**Comment by dam-ease** [comment]
Thanks for your response. 
Followed the steps again and the image still loads on my end, although a little slow which could be due to the network.
Looking through the issues you shared, I see this seems to be an issue and looks like a possible duplicate of https://github.com/flutter/flutter/issues/73942. Can you confirm if there is a difference between this issue and https://github.com/flutter/flutter/issues/73942??
If none, I'll suggest you upvote and comment on the original issue for more attention from the team.

**Comment by qtldleon** [comment]
It's same issue as https://github.com/flutter/flutter/issues/73942, but that issue is not resloved till now. 

Anyway, thanks for your update. I will try more in my end.

**Comment by dam-ease** [comment]
Thanks for your response.
Please upvote, comment and follow up on that issue for updates.
I'll be closing this in favour of the already existing issue.
Thank you!
